core/commands/publish.go

```diff
@@ -176,3 +181,36 @@ func publish(ctx context.Context, n *core.IpfsNode, k crypto.PrivKey, ref path.P
 		Value: ref.String(),
 	}, nil
 }
+
+func keylookup(k string, n *core.IpfsNode) (crypto.PrivKey, error) {
+
+	res, err := n.GetKey(k)
+	if res != nil {
```

Comment:
I would check here if error is different than `ErrNoSuchKey` if that is a case It signals different error than should happen here.